lib/palletjack/tool.rb

```diff
@@ -342,29 +342,42 @@ def pallet_dir(kind, name)
       config_dir :warehouse, kind, name
     end
 
-    # Write a key box file inside a pallet
+    # :call-seq:
+    # pallet_box kind, name, box, 'key.path' => value, ...
+    # pallet_box kind, name, box { { key: { path: value, ... } } }
+    #
+    # Write keys to a box file inside a pallet
+    #
+    # Any key.path assignments given as parameters, and the hash value
+    # returned from a block will be merged into the named box file.
     #
-    # The block should return a hash representing the contents of the box.
     # All keys will be stringified, so we can use key: short forms for
     # declaration of the box contents.
     #
     # Uses config_file to create the file, so any symbols will
     # be looked up in the options hash.
     #
-    # N.B! If the box already exists, it will be overwritten!
-    #--
-    # FIXME: should new values be merged with existing box data instead?
-    #++
-    #
     # Example:
     #
     #   pallet_box 'domain', :domain, 'dns' do
     #     { dns:{ ns:options[:soa_ns].split(',') } }
     #   end
+    #--
+    # FIXME: Box I/O should probably be managed by PalletJack::Pallet,
+    #        but that requires some redesign work.
+    #++
+
+    def pallet_box(kind, name, box, keys = {}, &block)
+      box = options[box] if box.is_a?(Symbol)
```

Comment:
Yes, but how is it supposed to be used? If the intention is to have a command line argument to specify the file name, I see a strong UI win in having something that can be tab-completed.